### PR TITLE
Lowercase internal variables in shell scripts

### PR DESCRIPTION
**Pull request by ulfalizer**
This makes it possible to tell at a glance which variables are internal to the script and which ones are parameters to it (or are exported from it), which is very helpful, and would've saved me a bunch of effort.

This convention is pretty common. See e.g. [Google's shell style guide](https://google.github.io/styleguide/shell.xml#Naming_Conventions), and [this random one](https://github.com/icy/bash-coding-style#variable-names). It's older than those though. Some other scripts in Zephyr already do it.

Also remove some semicolons at ends of lines.

(Some people do uppercase for "constants" as well, and leave lowercase to variables. Having some kind of distinction helps at least. I like the super simple lowercase-for-all-internal.)

### CODE REVIEW

**Comment by ulfalizer** [comment]
Some people do uppercase for "constants" as well, and leave lowercase to variables. Having some kind of distinction helps at least. I like the super simple lowercase-for-all-internal.

**Review by tyrantlizard77 — APPROVED** [review]

File: tests/bluetooth/bsim_bt/compile.sh [hunk 1]
```diff
@@ -19,38 +19,38 @@ BOARD_ROOT="${BOARD_ROOT:-${ZEPHYR_BASE}}"
...
+  local this_dir=${WORK_DIR}/${app}/${conf_file}
 
-  echo "Building $EXE_NAME"
+  echo "Building $exe_name"
 
-  #Set INCR_BUILD when calling to only do an incremental build
-  if [ ! -v INCR_BUILD ] || [ ! -d "${THIS_DIR}" ]; then
-      [ -d "${THIS_DIR}" ] && rm ${THIS_DIR} -rf
-      mkdir -p ${THIS_DIR} && cd ${THIS_DIR}
+  # Set incr_build when calling to only do an incremental build
```
Comment (aescolar) [inline tests/bluetooth/bsim_bt/compile.sh]: INCR_BUILD is be an optional input

**Review by aescolar — CHANGES_REQUESTED** [review]

File: tests/bluetooth/bsim_bt/compile.sh (hunk 1 above)
Comment (ulfalizer) [reply tests/bluetooth/bsim_bt/compile.sh]: Fixed

**Review by aescolar — APPROVED** [review]

**Comment by ulfalizer** [comment]
Rebased now.

**Pull request merged**